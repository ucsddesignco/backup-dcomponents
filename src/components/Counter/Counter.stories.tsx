import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import type { Meta, StoryObj } from "@storybook/react";
import Counter from "./Counter";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
  title: "DComponents/Counter",
  component: Counter,
  tags: ["autodocs"],
} satisfies Meta<typeof Counter>;

export default meta;
type Story = StoryObj<typeof meta>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Counter> = (args) => (
  <Counter {...args} />
);

export const Default: Story = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Default.args = { initialCount: 5 };
