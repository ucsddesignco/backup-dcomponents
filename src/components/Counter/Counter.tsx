import React, { useState } from "react";

export interface CounterProps {
  initialCount: number;
}

const Counter = ({ initialCount }: CounterProps) => {
  const [count, setCount] = useState(initialCount);
  return (
    <div>
      <p>Count: {count}</p>
      <button
        onClick={() => {
          setCount(count - 1);
        }}
      >
        -1
      </button>
      <button
        onClick={() => {
          setCount(count + 1);
        }}
      >
        +1
      </button>
    </div>
  );
};

export default Counter;
