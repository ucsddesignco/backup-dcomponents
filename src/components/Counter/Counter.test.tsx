import React from "react";
import { render } from "@testing-library/react";
import Counter from "./Counter";
import renderer from "react-test-renderer";

describe("Counter", () => {
  it("renders the Counter component", () => {
    render(<Counter initialCount={0} />);
  });
  it("renders the Counter component correctly with an initial count", () => {
    const tree = renderer.create(<Counter initialCount={0} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
