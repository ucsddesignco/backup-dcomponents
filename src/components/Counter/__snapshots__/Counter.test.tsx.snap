// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders the Counter component correctly with an initial count 1`] = `
<div>
  <p>
    Count: 
    0
  </p>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</div>
`;
